python/lsst/sims/catalogs/definitions/InstanceCatalog.py

```diff
@@ -254,6 +235,18 @@ def __init__(self, db_obj, obs_metadata=None, column_outputs=None,
                     if col not in self._column_outputs:
                         self._column_outputs.append(col)
 
+        self._cannot_be_null = None
```

Comment:
The three different forms of "cannot_be_null" makes this hard to understand at times if you lose track of which is which. Maybe adding a couple comments about what's going on here can help keep them straight.